src/icons/icon.tsx

```diff
@@ -0,0 +1,27 @@
+import { type SVGProps } from 'react';
+import spriteHref from './sprite.svg';
+import type { IconName } from './icon-constant';
+
+const defaultIconSizes = {
+  small: 18,
+  large: 24,
+};
+
+type DefaultIconSizes = keyof typeof defaultIconSizes;
+
+export function Icon({
+  name,
+  size = 'large',
+  ...props
+}: SVGProps<SVGSVGElement> & {
+  name: IconName;
+  size?: DefaultIconSizes | number;
+}) {
+  const actualSize = typeof size === 'string' ? defaultIconSizes[size] : size;
```

Comment:
let's use a type guard instead

```suggestion

const isDefaultIconSize = (size: unknown): size is DefaultIconSizes => size === 'small' || size === 'large';

export function Icon({
  name,
  size = 'large',
  ...props
}: SVGProps<SVGSVGElement> & {
  name: IconName;
  size?: DefaultIconSizes | number;
}) {
  const actualSize = isDefaultIconSize(size) ? defaultIconSizes[size] : size;
```

Reply:
@cfernando415 Using a type predicate here might be counterproductive.

Consider this example:

```ts
<Icon size="somestring" name="arrow"/>
```

If a type predicate is used, `actualSize` will be set to `somestring`, which is invalid. However, with the existing method, `defaultIconSizes['somestring']` results in `undefined`, making `actualSize` also undefined (a valid entry).

This issue is already prevented by the type of size, unless there's a type casting. But the present method ensures both type-safety and runtime safety.

Please let me know what you think.

Reply:
hmmm.. thanks for explaining! I understand the intention 😄 

resolving this thread - no change needed here